### PR TITLE
[release-v1.4] Tracing tests for Kafka Broker

### PR DESCRIPTION
**Pull request by mgencur**
Backport of upstream https://github.com/knative-sandbox/eventing-kafka-broker/pull/2454

Changes:
* Pull the relevant commits from upstream
* Add OpenShift specific files for setting up Zipkin/Tracing

### CODE REVIEW

File: openshift/e2e-common.sh [hunk 1]
```diff
@@ -90,10 +90,16 @@ EOF
...
   popd || return $?
 
-  oc apply -f openshift/knative-eventing.yaml
+  local eventing_cr="$(mktemp -t eventing-XXXXX.yaml)"
+  cp openshift/knative-eventing.yaml "$eventing_cr"
+  enable_tracing "$eventing_cr"
+  oc apply -f "$eventing_cr"
+
+  # Install KnativeKafka after installing KnativeEventing with tracing enabled
+  oc apply -f openshift/knative-kafka.yaml
```
Comment (pierDipi) [inline openshift/e2e-common.sh:102]: Can we use the operator to deploy everything?

Comment (pierDipi) [reply openshift/e2e-common.sh:102]: with something like:
```
export TRACING_BACKEND=zipkin
OPENSHIFT_CI="true" make generated-files install-tracing install-kafka || failed=$?
```
I would expect it to work.

Comment (mgencur) [reply openshift/e2e-common.sh:102]: The problem is that the knative-eventing custom resource from serverless-operator is different from knative-eventing in eventing-kafka-broker. The difference is this: https://github.com/openshift-knative/eventing-kafka-broker/blob/release-v1.4/openshift/knative-eventing.yaml#L7-L15 However, maybe I could add these lines via patch after knative-eventing and knative-kafka are installed? Will this cause any issues?

Comment (mgencur) [reply openshift/e2e-common.sh]: OK. Let's try it, pushed a new commit.

Comment (mgencur) [reply openshift/e2e-common.sh]: It doesn't work. The config-tracing configmap is empty. Will look into it tomorrow.

Comment (mgencur) [reply openshift/e2e-common.sh]: So. Pushed another commit. Now I'm re-using the tracing and KnativeKafka installation from serverless-operator. And I'm installing a specific KnativeEventing from eventing-kafka-broker repo.

**Review by pierDipi — APPROVED** [review]
Thank you!

/lgtm

**Comment by mgencur** [comment]
@pierDipi cool, I'm glad we found a solution that works for both sides :-D Thanks for the review.

**Comment by pierDipi** [comment]
@mgencur I will bring `openshift/*` changes to main myself once https://github.com/openshift-knative/eventing-kafka-broker/pull/359 is in

**Pull request merged**